### PR TITLE
Reorganized to DMIParse and DMIDecode objects

### PR DESCRIPTION
**Pull request by luboss**
Code reorganization and new features

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,36 @@
...
 
 ## How to use
 ```python
-import dmidecode
+from dmidecode import DMIDecode
+import subprocess
 
-# parse the full output
-data = dmidecode.parse()
+# run dmidecode and store its output
```
Comment (zaibon) [inline README.md]: I would move these 2 line to the line 29 in the second example where you show how to use the DMIParse object.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
+Alternatively instead of running dmidecode locally you can use DMIParse by passing dmidecode output as an argument:
+```python
+from dmidecode import DMIParse
+dmi = DMIParse(raw)
+```
+
+Other information can be easily retrieved by analyzing dmi.data and module code.
+
+## Possible limitations
+Tested with dmidecode versions 2.11 and 2.12 only
```
Comment (zaibon) [inline README.md]: I also tested with dmidecode version 3.2, so I think it is safe to add it to the list of tested versions

File: README.md (hunk 1 above)
Comment (luboss) [reply README.md]: Oh, I forgot to remove that part. Running dmidecode is already covered in DMIDecode's __init__(). Removing...

**Comment by luboss** [comment]
Christophe, I'm kind of lost regarding black. What is wrong with my formatting? I'm following PEP-8 style using flake8. Where can I find mentioned non-green github actions?

**Comment by zaibon** [comment]
Nothing is wrong with it by itself. I just choose to use black to format all the code so I can just ensure that the same styling is used everywhere. You can read about it at: https://github.com/psf/black

Regarding the github actions, not sure why, but they don't seem to be triggered in your PR. Maybe because I added them after you opened your PR.

**Comment by zaibon** [comment]
The content of your PR is correct. So don't bother with black I can always just format the code once after I merge. 
Just make sure the conflict is resolved and I will merge it.

**Comment by luboss** [comment]
Actually I do not use any Github actions at all (if you are referring to [this](https://github.com/features/actions).
Regarding black, I wanted to install python-black as instructed in the Atom plugin (editor I use) but pip could not find it (tried both 3.6 and 2.7).
I resolved the conflict by putting my version of dmidecode.py. If you want to change/add anything, you can do it after pull to master.

**Comment by zaibon** [comment]
@luboss all merged ! Thanks for your contribution :-)

**Comment by luboss** [comment]
Thank you too @zaibon for pulling my changes and support. I'm happy to see the new major release (0.1.0) of py-dmidecode at pypi.org :-)

**Pull request merged**